faiss/impl/RaBitQuantizer.cpp

```diff
@@ -0,0 +1,487 @@
+#include <faiss/impl/RaBitQuantizer.h>
+
+#include <algorithm>
+#include <cmath>
+#include <cstring>
+#include <limits>
+#include <memory>
+#include <vector>
+
+#include <faiss/impl/FaissAssert.h>
+#include <faiss/impl/platform_macros.h>
+#include <faiss/utils/distances.h>
+
+namespace faiss {
+
+struct FactorsData {
+    // ||or - c||
+    float factor_0 = 0;
+    // sum_xb
+    float factor_1 = 0;
+    float factor_2 = 0;
+    // ||or||^2
+    float factor_3 = 0;
+};
```

Comment:
In my implementation there are only 2 floats per database vector. Do you store additional ones for efficiency?